test/esockd_SUITE.erl

```diff
@@ -456,3 +490,39 @@ reuse_socket(_) ->
     end,
     esockd_transport:close(Sock1).
 
+connection_sup_api(_Config) ->
+    {ok, LSup} = esockd:open(esockd_connect_SUITE, {"127.0.0.1", 1314},
+                              ?OPTS, echo_mfa()),
+    ConnSup = esockd_listener_sup:connection_sup(LSup),
+    MaxConn = esockd_connection_sup:get_max_connections(ConnSup),
+    ?assertEqual(MaxConn, 1024000),
+
+    esockd_connection_sup:set_max_connections(ConnSup, 2),
+    RestMaxConn = esockd_connection_sup:get_max_connections(ConnSup),
+    ?assertEqual(RestMaxConn, 2),
+
+    {ok, Sock1} = gen_tcp:connect({127,0,0,1}, 1314, []),
+    timer:sleep(10),
+    CurrentConn = esockd_connection_sup:count_connections(ConnSup),
+    ?assertEqual(CurrentConn, 1),
+    esockd_transport:close(Sock1),
+    timer:sleep(10),
+
+    [{Reason, ShutdownConn}] = esockd_connection_sup:get_shutdown_count(ConnSup),
```

Comment:
Should we check the value of 'Reason'? Or just ignore the value?
You could install some kind of `erlang-linters` plugin in your editor, to check and info you the compile warnings.